Deis logs "panic: runtime error: index out of range"

Calling `deis logs` on these lines (taken from `docker logs`):
```
[2015-09-06T13:37:19.517Z] Preparing production files
[2015-09-06T13:37:22.190Z] Done preparing production files with null / "\u001b[4mRunning \"plugins\" task\u001b[24m\nPreparing production files for following plugins: [\"plugins\",\"density\",\"locale\",\"enterpriseinfo\",\"push\"]\nDone preparing production files\n\n\u001b[4mRunning \"concat:plugins\" (concat) task\u001b[24m\nFile frontend/express/public/javascripts/min/countly.plugins.concat.js created.\n\n\u001b[4mRunning \"uglify:plugins\" (uglify) task\u001b[24m\n\u001b[32m>> \u001b[39m1 file created.\n\n\u001b[4mRunning \"copy:plugins\" (copy) task\u001b[24m\nCopied 1 file\n\n\u001b[4mRunning \"cssmin:plugins\" (cssmin) task\u001b[24m\n\u001b[32m>> \u001b[39m1 file created. 16.57 kB → 13.74 kB\n\n\u001b[4mRunning \"locales\" task\u001b[24m\nPreparing locale files for core & plugins: [\"plugins\",\"density\",\"locale\",\"enterpriseinfo\",\"push\"]\nDone preparing locale files\n\n\u001b[4mRunning \"concat:locales_\" (concat) task\u001b[24m\nFile frontend/express/public/localization/min/locale.properties created.\n\n\u001b[4mRunning \"concat:locales_ar\" (concat) task\u001b[24m\nFile frontend/express/public/localization/min/locale_ar.properties created.\n\n\u001b[4mRunning \"concat:locales_ca\" (concat) task\u001b[24m\nFile frontend/express/public/localization/min/locale_ca.properties created.\n\n\u001b[4mRunning \"concat:locales_de\" (concat) task\u001b[24m\nFile frontend/express/public/localization/min/locale_de.properties created.\n\n\u001b[4mRunning \"concat:locales_el\" (concat) task\u001b[24m\nFile frontend/express/public/localization/min/locale_el.properties created.\n\n\u001b[4mRunning \"concat:locales_en\" (concat) task\u001b[24m\nFile frontend/express/public/localization/min/locale_en.properties created.\n\n\u001b[4mRunning \"concat:locales_es\" (concat) task\u001b[24m\nFile frontend/express/public/localization/min/locale_es.properties created.\n\n\u001b[4mRunning \"concat:locales_et\" (concat) task\u001b[24m\nFile frontend/express/public/localization/min/locale_et.properties created.\n\n\u001b[4mRunning \"concat:locales_fr\" (concat) task\u001b[24m\nFile frontend/express/public/localization/min/locale_fr.properties created.\n\n\u001b[4mRunning \"concat:locales_it\" (concat) task\u001b[24m\nFile frontend/express/public/localization/min/locale_it.properties created.\n\n\u001b[4mRunning \"concat:locales_ja\" (concat) task\u001b[24m\nFile frontend/express/public/localization/min/locale_ja.properties created.\n\n\u001b[4mRunning \"concat:locales_ko\" (concat) task\u001b[24m\nFile frontend/express/public/localization/min/locale_ko.properties created.\n\n\u001b[4mRunning \"concat:locales_lv\" (concat) task\u001b[24m\nFile frontend/express/public/localization/min/locale_lv.properties created.\n\n\u001b[4mRunning \"concat:locales_nb\" (concat) task\u001b[24m\nFile frontend/express/public/localization/min/locale_nb.properties created.\n\n\u001b[4mRunning \"concat:locales_nl\" (concat) task\u001b[24m\nFile frontend/express/public/localization/min/locale_nl.properties created.\n\n\u001b[4mRunning \"concat:locales_pt\" (concat) task\u001b[24m\nFile frontend/express/public/localization/min/locale_pt.properties created.\n\n\u001b[4mRunning \"concat:locales_ro\" (concat) task\u001b[24m\nFile frontend/express/public/localization/min/locale_ro.properties created.\n\n\u001b[4mRunning \"concat:locales_ru\" (concat) task\u001b[24m\nFile frontend/express/public/localization/min/locale_ru.properties created.\n\n\u001b[4mRunning \"concat:locales_sl\" (concat) task\u001b[24m\nFile frontend/express/public/localization/min/locale_sl.properties created.\n\n\u001b[4mRunning \"concat:locales_sv\" (concat) task\u001b[24m\nFile frontend/express/public/localization/min/locale_sv.properties created.\n\n\u001b[4mRunning \"concat:locales_tr\" (concat) task\u001b[24m\nFile frontend/express/public/localization/min/locale_tr.properties created.\n\n\u001b[4mRunning \"concat:locales_vi\" (concat) task\u001b[24m\nFile frontend/express/public/localization/min/locale_vi.properties created.\n\n\u001b[4mRunning \"concat:locales_zh\" (concat) task\u001b[24m\nFile frontend/express/public/localization/min/locale_zh.properties created.\n\n\u001b[4mRunning \"concat:locales_pl\" (concat) task\u001b[24m\nFile frontend/express/public/localization/min/locale_pl.properties created.\n\n\u001b[32mDone, without errors.\u001b[39m\n"
```

... causes this error:

```
2015-09-06T13:37:19UTC cly[cmd.1]: [2015-09-06T13:37:19.517Z] Preparing production files
2015-09-06T13:37:22UTC cly[cmd.1]: [2015-09-06T13:37:22.190Z] Done preparing production files with null / \"\\u001b[4mRunning \\\"plugins\\\" task\\u001b[24m\
Preparing production files for following plugins: [\\\"plugins\\\",\\\"density\\\",\\\"locale\\\",\\\"enterpriseinfo\\\",\\\"push\\\"]\
Done preparing production files\
panic: runtime error: index out of range

goroutine 1 [running]:
github.com/deis/deis/client/cmd.AppLogs(0x8207901e1, 0x3, 0xffffffffffffffff, 0x0, 0x0)
	/Users/matt/Projects/src/github.com/deis/deis/client/cmd/apps.go:162 +0x571
github.com/deis/deis/client/parser.appLogs(0x8206e61d0, 0x1, 0x1, 0x0, 0x0)
	/Users/matt/Projects/src/github.com/deis/deis/client/parser/apps.go:192 +0x2b6
github.com/deis/deis/client/parser.Apps(0x8206e61d0, 0x1, 0x1, 0x0, 0x0)
	/Users/matt/Projects/src/github.com/deis/deis/client/parser/apps.go:37 +0x3e9
main.Command(0x8206e61d0, 0x1, 0x1, 0x186840)
	/Users/matt/Projects/src/github.com/deis/deis/client/deis.go:87 +0xc24
main.main()
	/Users/matt/Projects/src/github.com/deis/deis/client/deis.go:18 +0x60
```

I'm on Deis 1.10.0.

--- a/client/cmd/apps.go
+++ b/client/cmd/apps.go
@@ -158,10 +158,19 @@ func AppLogs(appID string, lines int) error {
 		return err
 	}
 
+	return printLogs(logs)
+}
+
+// printLogs prints each log line with a color matched to its category.
+func printLogs(logs string) error {
 	for _, log := range strings.Split(strings.Trim(logs, `\n`), `\n`) {
-		catagory := strings.Split(strings.Split(log, ": ")[0], " ")[1]
+		category := "unknown"
+		parts := strings.Split(strings.Split(log, ": ")[0], " ")
+		if len(parts) >= 2 {
+			category = parts[1]
+		}
 		colorVars := map[string]string{
-			"Color": chooseColor(catagory),
+			"Color": chooseColor(category),
 			"Log":   log,
 		}
 		fmt.Println(prettyprint.ColorizeVars("{{.V.Color}}{{.V.Log}}{{.C.Default}}", colorVars))

--- a/client/cmd/apps_test.go
+++ b/client/cmd/apps_test.go
@@ -1,0 +1,18 @@
+package cmd
+
+import "testing"
+
+func TestPrintLogLinesBadLine(t *testing.T) {
+	t.Parallel()
+
+	// Regression test for https://github.com/deis/deis/issues/4420
+	logs := `\nDone preparing production files\n\n\u001b[4mRunning \"concat:plugins\" (concat) task\u001b[24m\n`
+	if err := printLogs(logs); err != nil {
+		t.Fatal(err)
+	}
+
+	logs = `\n\n\n`
+	if err := printLogs(logs); err != nil {
+		t.Fatal(err)
+	}
+}

